### PR TITLE
Added ENABLE_CEREAL option with default OFF

### PR DESCRIPTION
**Pull request by keceli**
When this option is on, cereal is downloaded and installed.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -66,6 +66,7 @@ set(MADNESS_INSTALL_CMAKEDIR "lib/cmake/madness"
 # Build options  ==============================================================
 
 redefaultable_option(MADNESS_BUILD_MADWORLD_ONLY "Build only MADWorld runtime (excluding BLAS/LAPACK interfaces)" OFF)
+redefaultable_option(ENABLE_CEREAL "Use Cereal library along with MADNESS archive for serialization" OFF)
```
Comment (evaleev) [inline CMakeLists.txt:69]: should rename to make it MADNESS-specific (`MADNESS_ENABLE_CEREAL`)

File: cmake/modules/FindCereal.cmake [hunk 2]
```diff
@@ -0,0 +1,19 @@
+find_package( cereal QUIET )
+if( cereal_FOUND )
+    target_compile_definitions(cereal INTERFACE
+      "CEREAL_THREAD_SAFE=1;GAUXC_ENABLE_CEREAL=1;JUST_INSTALL_CEREAL=ON;SKIP_PORTABILITY_TEST=ON")
+else()
+    message( STATUS "Could not find Cereal. Downloading..." )
+    include(FetchContent)
+    FetchContent_Declare(
+      cereal
+      GIT_REPOSITORY git@github.com:USCiLab/cereal.git)
```
Comment (evaleev) [inline cmake/modules/FindCereal.cmake:10]: need to pin to specific commit

**Review by evaleev — CHANGES_REQUESTED** [review]
since now MADNESS tests for presence of cereal should define macro and use that to enable cereal-based archives rather than use `__has_include` (the latter can trigger even if `MADNESS_ENABLE_CEREAL=OFF`)

File: cmake/modules/FindCereal.cmake (hunk 2 above)
Comment (keceli) [reply cmake/modules/FindCereal.cmake]: Done with 7412c09.

File: CMakeLists.txt (hunk 1 above)
Comment (keceli) [reply CMakeLists.txt]: Done with 7412c09.

**Comment by keceli** [comment]
> since now MADNESS tests for presence of cereal should define macro and use that to enable cereal-based archives rather than use `__has_include` (the latter can trigger even if `MADNESS_ENABLE_CEREAL=OFF`)

Thank you @evaleev for reviewing. I added `add_compile_definitions(MADNESS_HAS_CEREAL)` to `FindCereal.cmake` and set an error if `MADNESS_ENABLE_CEREAL=ON` but cereal header is not found.

File: cmake/modules/FindCereal.cmake [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    FetchContent_GetProperties(cereal)
+    if(NOT cereal_POPULATED)
+        FetchContent_Populate( cereal )
+        add_library( cereal INTERFACE IMPORTED )
+        set_target_properties( cereal PROPERTIES
+          INTERFACE_INCLUDE_DIRECTORIES "${cereal_SOURCE_DIR}/include"
+          INTERFACE_COMPILE_DEFINITIONS "CEREAL_THREAD_SAFE=1;GAUXC_ENABLE_CEREAL=1;JUST_INSTALL_CEREAL=ON;SKIP_PORTABILITY_TEST=ON")
+  endif()
+endif()
+add_compile_definitions(MADNESS_HAS_CEREAL)
```
Comment (evaleev) [inline cmake/modules/FindCereal.cmake:21]: old-school global definitions like these should be avoided in modern CMake. Either needs to go to madness_config.h or better defined for MADworld and madness targets

**Review by evaleev — CHANGES_REQUESTED** [review]

File: cmake/modules/FindCereal.cmake (hunk 3 above)
Comment (keceli) [reply cmake/modules/FindCereal.cmake]: Would this abb7c1c2 be fine?

File: cmake/modules/FindCereal.cmake [hunk 4]
```diff
@@ -1,20 +0,0 @@
...
-      cereal
-      GIT_REPOSITORY git@github.com:USCiLab/cereal.git
-      GIT_TAG v1.3.0)
-    FetchContent_GetProperties(cereal)
-    if(NOT cereal_POPULATED)
-        FetchContent_Populate( cereal )
-        add_library( cereal INTERFACE IMPORTED )
-        set_target_properties( cereal PROPERTIES
-          INTERFACE_INCLUDE_DIRECTORIES "${cereal_SOURCE_DIR}/include"
-          INTERFACE_COMPILE_DEFINITIONS "CEREAL_THREAD_SAFE=1;GAUXC_ENABLE_CEREAL=1;JUST_INSTALL_CEREAL=ON;SKIP_PORTABILITY_TEST=ON")
```
Comment (evaleev) [inline cmake/modules/FindCereal.cmake]: @keceli it's not Cereal's find module to be aware of other modules. respect encapsulation. Gauxc needs to check for presence of cereal target and configure itself appropriately.

**Review by evaleev — APPROVED** [review]

**Comment by evaleev** [comment]
@keceli after your changes CI never actually used cereal ... please fix, still not working for some reason

**Comment by keceli** [comment]
> @keceli after your changes CI never actually used cereal ... please fix, still not working for some reason

7de6f6e should fix the problem, but reverts some of the changes you made in dbeaacb.

**Pull request merged**